bin/ch/Debugger.cpp

```diff
@@ -0,0 +1,526 @@
+
+    JsValueRef args[] = { undefinedValue };
+    JsValueRef result = JS_INVALID_REFERENCE;
+    IfJsrtErrorFailLogAndRetFalse(ChakraRTInterface::JsCallFunction(globalFunc, args, _countof(args), &result));
+
+    JsValueRef globalObj = JS_INVALID_REFERENCE;
+    IfJsrtErrorFailLogAndRetFalse(ChakraRTInterface::JsGetGlobalObject(&globalObj));
+
+    JsPropertyIdRef hostDebugObjectPropId;
+    IfJsrtErrorFailLogAndRetFalse(ChakraRTInterface::JsGetPropertyIdFromName(_u("hostDebugObject"), &hostDebugObjectPropId));
+
+    JsPropertyIdRef hostDebugObject;
+    IfJsrtErrorFailLogAndRetFalse(ChakraRTInterface::JsGetProperty(globalObj, hostDebugObjectPropId, &hostDebugObject));
+
+    this->InstallDebugCallbacks(hostDebugObject);
+
+    if (!WScriptJsrt::Initialize())
+    {
+        return false;
+    }
+
+    if (HostConfigFlags::flags.dbgbaselineIsEnabled)
+    {
+        this->SetBaseline();
+    }
+
+    if (HostConfigFlags::flags.InspectMaxStringLengthIsEnabled)
+    {
+        this->SetInspectMaxStringLength();
+    }
+
+    return true;
+}
+
+bool Debugger::InstallDebugCallbacks(JsValueRef hostDebugObject)
+{
+    HRESULT hr = S_OK;
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetSource"), Debugger::JsDiagGetSource));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagSetBreakpoint"), Debugger::JsDiagSetBreakpoint));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetStackTrace"), Debugger::JsDiagGetStackTrace));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetBreakpoints"), Debugger::JsDiagGetBreakpoints));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagRemoveBreakpoint"), Debugger::JsDiagRemoveBreakpoint));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagSetBreakOnException"), Debugger::JsDiagSetBreakOnException));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetBreakOnException"), Debugger::JsDiagGetBreakOnException));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagSetStepType"), Debugger::JsDiagSetStepType));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetScripts"), Debugger::JsDiagGetScripts));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetStackProperties"), Debugger::JsDiagGetStackProperties));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetProperties"), Debugger::JsDiagGetProperties));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagGetObjectFromHandle"), Debugger::JsDiagGetObjectFromHandle));
+    IfFalseGo(WScriptJsrt::InstallObjectsOnObject(hostDebugObject, _u("JsDiagEvaluate"), Debugger::JsDiagEvaluate));
+Error:
+    return hr != S_OK;
+}
+
+bool Debugger::SetBaseline()
```

Comment:
NIT: I don't see a reason to return bool, I don't see you are checking the return value.